File: bigwater-backend/quarkus-api/src/main/java/com/app6768688/config/ApiUrls.java
package com.app6768688.config;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;
import org.eclipse.microprofile.config.inject.ConfigProperty;

/**
 * Centralized API URLs configuration
 * This class contains all external API URLs used throughout the application
 */
@ApplicationScoped
public class ApiUrls {
    
    // Blockchain API URLs
    @Inject
    @ConfigProperty(name = "bw.api.polygon.url", defaultValue = "https://api.polygonscan.com/api")
    String polygonApiUrl;
    
    @Inject
    @ConfigProperty(name = "bw.api.tron.url", defaultValue = "https://api.trongrid.io")
    String tronApiUrl;
    
    @Inject
    @ConfigProperty(name = "bw.api.polygon.amoy.url", defaultValue = "https://api-amoy.polygonscan.com/api")
    String polygonAmoyApiUrl;
    
    @Inject
    @ConfigProperty(name = "bw.api.alchemy.polygon.url", defaultValue = "https://polygon-mainnet.g.alchemy.com/v2")
    String alchemyPolygonUrl;
    
    @Inject
    @ConfigProperty(name = "bw.api.moralis.polygon.url", defaultValue = "https://deep-index.moralis.io/api/v2")
    String moralisPolygonUrl;
    
    // Getters for API URLs
    public String getPolygonApiUrl() {
        return polygonApiUrl;
    }
    
    public String getTronApiUrl() {
        return tronApiUrl;
    }
    
    public String getPolygonAmoyApiUrl() {
        return polygonAmoyApiUrl;
    }
    
    public String getAlchemyPolygonUrl() {
        return alchemyPolygonUrl;
    }
    
    public String getMoralisPolygonUrl() {
        return moralisPolygonUrl;
    }
    
    /**
     * Get API URL by service and network
     * @param service Service name (polygon, tron, alchemy, moralis)
     * @param network Network name (mainnet, amoy, etc.)
     * @return API URL or null if not found
     */
    public String getApiUrl(String service, String network) {
        if ("polygon".equalsIgnoreCase(service)) {
            if ("amoy".equalsIgnoreCase(network)) {
                return polygonAmoyApiUrl;
            }
            return polygonApiUrl;
        } else if ("tron".equalsIgnoreCase(service)) {
            return tronApiUrl;
        } else if ("alchemy".equalsIgnoreCase(service)) {
            return alchemyPolygonUrl;
        } else if ("moralis".equalsIgnoreCase(service)) {
            return moralisPolygonUrl;
        }
        return null;
    }
    
    /**
     * Get all API URLs as a map
     * @return Map of service names to URLs
     */
    public java.util.Map<String, String> getAllApiUrls() {
        java.util.Map<String, String> urls = new java.util.HashMap<>();
        urls.put("polygon", polygonApiUrl);
        urls.put("tron", tronApiUrl);
        urls.put("polygon_amoy", polygonAmoyApiUrl);
        urls.put("alchemy_polygon", alchemyPolygonUrl);
        urls.put("moralis_polygon", moralisPolygonUrl);
        return urls;
    }
}

File: bigwater-ui/vue3/src/utils/apiUrls.js
/**
 * Centralized API URLs configuration
 * This file contains all external API URLs used throughout the frontend
 */

// Backend API URLs
export const BACKEND_API_URLS = {
  // Base API URL
  BASE: process.env.NODE_ENV === 'production' 
    ? 'https://your-production-api.com/bw-api' 
    : 'http://localhost:9999/bw-api',
  
  // Authentication endpoints
  LOGIN: '/auth/login',
  LOGOUT: '/auth/logout',
  REGISTER: '/auth/register',
  VALIDATE: '/auth/validate',
  
  // Wallet endpoints
  WALLETS: '/wallets',
  WALLETS_USER: '/wallets/user',
  WALLETS_COMPANY: '/wallets/company',
  WALLETS_COMPANY_RANDOM: '/wallets/company/random',
  
  // Transaction endpoints
  TRANSACTIONS: '/transactions',
  TRANSACTIONS_USER: '/transactions/user',
  TRANSACTIONS_VERIFY: '/transactions/verify',
  TRANSACTIONS_VERIFY_TEST: '/transactions/verify/test',
  
  // Payment endpoints
  RECORD_PAYMENT: '/transactions/record-payment'
}

// External blockchain API URLs (for reference)
export const BLOCKCHAIN_API_URLS = {
  // Polygon APIs
  POLYGON_SCAN: 'https://api.polygonscan.com/api',
  POLYGON_AMOY_SCAN: 'https://api-amoy.polygonscan.com/api',
  
  // TRON APIs
  TRON_GRID: 'https://api.trongrid.io',
  
  // Alchemy APIs
  ALCHEMY_POLYGON: 'https://polygon-mainnet.g.alchemy.com/v2',
  
  // Moralis APIs
  MORALIS_POLYGON: 'https://deep-index.moralis.io/api/v2'
}

/**
 * Get full API URL by combining base URL with endpoint
 * @param {string} endpoint API endpoint
 * @returns {string} Full API URL
 */
export function getApiUrl(endpoint) {
  return BACKEND_API_URLS.BASE + endpoint
}

/**
 * Get all backend API URLs
 * @returns {Object} Map of endpoint names to full URLs
 */
export function getAllBackendUrls() {
  const urls = {}
  for (const [key, value] of Object.entries(BACKEND_API_URLS)) {
    if (key !== 'BASE') {
      urls[key] = getApiUrl(value)
    }
  }
  return urls
}

/**
 * Get blockchain API URL by service and network
 * @param {string} service Service name (polygon, tron, alchemy, moralis)
 * @param {string} network Network name (mainnet, amoy, etc.)
 * @returns {string|null} API URL or null if not found
 */
export function getBlockchainApiUrl(service, network) {
  if (service === 'polygon') {
    if (network === 'amoy') {
      return BLOCKCHAIN_API_URLS.POLYGON_AMOY_SCAN
    }
    return BLOCKCHAIN_API_URLS.POLYGON_SCAN
  } else if (service === 'tron') {
    return BLOCKCHAIN_API_URLS.TRON_GRID
  } else if (service === 'alchemy') {
    return BLOCKCHAIN_API_URLS.ALCHEMY_POLYGON
  } else if (service === 'moralis') {
    return BLOCKCHAIN_API_URLS.MORALIS_POLYGON
  }
  return null
}
